### PR TITLE
Fixed the three dots menu issue on project-dashboard

### LINKED ISSUE #591: BUG: Adjust 3 dot menu for board cards
### Bug
**Overview**
The three-dot menu no longer displays in the top right corner of the card. 
Screenshot 2024-06-24 at 9 25 54 AM

**To Reproduce**
Steps to reproduce:
1. Create an account and login
2. On the My Projects page, click the plus button in the right of the nav bar to create a new project
Screenshot 2024-06-24 at 9 27 04 AM

3. On the right of the boards dashboard, click the plus button to create a board
Screenshot 2024-06-24 at 9 28 27 AM

4. See that the more_vert (3-dot) menu icon is displayed in the middle-right of the card rather than the top-right. 

**Proposed tasks to fix**
- [x] You may want to look at the menu icon for projects on the My Projects page to get some ideas (or improve on those ideas) for how to implement this change.
- [x] Change the styling to ensure the three-dot menu is in the top right
- [x] Ensure that there is sufficient padding between board name and the menu for long menu names (for example try this name "Test Board long name ..." and ensure there is padding between the text and the menu).

**Screenshots**
Boards dashboard
Screenshot 2024-06-24 at 9 33 58 AM

Projects dashboard
Screenshot 2024-06-24 at 9 34 33 AM

### PR DESCRIPTION
**Pull request by markiianbabiak**
Moved the three dots menu from the mat-card into mat-card-title

Closes #591

### CODE REVIEW

**Comment by JoelWiebe** [comment]
@marieklinaeva would you mind reviewing this as well -- to go through the process of moving between branches and build a habit of us all reviewing each other's work :)

Checkout clicking this above if you have GitHub Desktop installed:
Screenshot 2024-06-25 at 12 00 13 PM

File: frontend/src/app/components/project-dashboard/project-dashboard.component.html [hunk 1]
```diff
@@ -110,18 +110,7 @@ <h3 class="mat-title" *ngIf="project">Boards for {{ project.name }}</h3>
...
-                  class="accent"
-                  *ngIf="user && user.role === Role.TEACHER"
-                  matTooltip="Toggle Visibility"
-                  (click)="toggleBoardVisibility($event, board)"
-                >
-                  <mat-icon *ngIf="board.visible">visibility</mat-icon>
-                  <mat-icon *ngIf="!board.visible">visibility_off</mat-icon>
-                </button>
-                <button 
+                  <button
```
Comment (JoelWiebe) [inline frontend/src/app/components/project-dashboard/project-dashboard.component.html:113]: Amazing! You made quick work of this. Just adjust the indentation here, then looks good to me. Let's wait for this commit and Marie's review then we can merge.

Looks like I'll need to find something more challenging for you!

**Review by JoelWiebe — COMMENTED** [review]
That was quick! Just one indentation adjustment, otherwise, looks good to me :)

**Review by marieklinaeva — COMMENTED** [review]
I'm still learning the code, but it looks good to me.

**Review by JoelWiebe — APPROVED** [review]
Great work, feel free to merge!

**Pull request merged**